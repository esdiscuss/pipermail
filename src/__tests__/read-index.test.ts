import readIndex from '../read-index';

test('parses the index to return an array of month urls', async () => {
  const res = await readIndex('https://mail.mozilla.org/pipermail/es-discuss/');
  expect(Array.isArray(res)).toBe(true);
  expect(res.length).toBeGreaterThan(70);
  res.forEach(url => {
    expect(url).toMatch(
      /^https:\/\/mail.mozilla.org\/pipermail\/es-discuss\/\d\d\d\d\-[A-Z][a-z]+$/,
    );
  });
  // by only looking at the oldest months, they should rarely change
  expect(res.slice(0, 70)).toMatchSnapshot();
});
