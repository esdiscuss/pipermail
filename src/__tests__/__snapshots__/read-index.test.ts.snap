// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses the index to return an array of month urls 1`] = `
Array [
  "https://mail.mozilla.org/pipermail/es-discuss/2006-June",
  "https://mail.mozilla.org/pipermail/es-discuss/2006-July",
  "https://mail.mozilla.org/pipermail/es-discuss/2006-August",
  "https://mail.mozilla.org/pipermail/es-discuss/2006-September",
  "https://mail.mozilla.org/pipermail/es-discuss/2006-October",
  "https://mail.mozilla.org/pipermail/es-discuss/2006-November",
  "https://mail.mozilla.org/pipermail/es-discuss/2006-December",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-January",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-February",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-March",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-April",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-May",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-June",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-July",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-August",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-September",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-October",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-November",
  "https://mail.mozilla.org/pipermail/es-discuss/2007-December",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-January",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-February",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-March",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-April",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-May",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-June",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-July",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-August",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-September",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-October",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-November",
  "https://mail.mozilla.org/pipermail/es-discuss/2008-December",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-January",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-February",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-March",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-April",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-May",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-June",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-July",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-August",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-September",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-October",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-November",
  "https://mail.mozilla.org/pipermail/es-discuss/2009-December",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-January",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-February",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-March",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-April",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-May",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-June",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-July",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-August",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-September",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-October",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-November",
  "https://mail.mozilla.org/pipermail/es-discuss/2010-December",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-January",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-February",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-March",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-April",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-May",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-June",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-July",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-August",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-September",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-October",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-November",
  "https://mail.mozilla.org/pipermail/es-discuss/2011-December",
  "https://mail.mozilla.org/pipermail/es-discuss/2012-January",
  "https://mail.mozilla.org/pipermail/es-discuss/2012-February",
  "https://mail.mozilla.org/pipermail/es-discuss/2012-March",
]
`;
